homeassistant/components/media_player/soundtouch.py

```diff
@@ -70,180 +70,99 @@
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup the Bose Soundtouch platform."""
-    name = config.get(CONF_NAME)
-
-    remote_config = {
-        'name': 'HomeAssistant',
-        'description': config.get(CONF_NAME),
-        'id': 'ha.component.soundtouch',
-        'port': config.get(CONF_PORT),
-        'host': config.get(CONF_HOST)
-    }
-
-    soundtouch_device = SoundTouchDevice(name, remote_config)
-    DEVICES.append(soundtouch_device)
-    add_devices([soundtouch_device])
+    if DATA_SOUNDTOUCH not in hass.data:
+        hass.data[DATA_SOUNDTOUCH] = []
+
+    if discovery_info:
+        # Discovery
+        host = discovery_info[0]
```

Comment:
Discovery info is a dictionary. Make sure your local dev environment has the latest `netdisco` (as required by the latest HASS)